external/fv3fit/fv3fit/_shared/packer.py

```diff
@@ -120,9 +127,16 @@ def pack(
 
     data_clipped = xr.Dataset(clip(data, config.clip))
     stacked = data_clipped.to_stacked_array(feature_dim_name, sample_dims=sample_dims)
-    stacked = stacked.dropna(feature_dim_name)
+    if CUPY_AVAILABLE and stacked.cupy.is_cupy:
+        # dropna features a numpy function which breaks on device cupy expectations
+        stacked = stacked.cupy.as_numpy()
+        stacked = stacked.dropna(feature_dim_name)
+        stacked = stacked.cupy.as_cupy()
```

Comment:
It looks like we can use [`apply_ufunc`](https://cupy-xarray.readthedocs.io/source/apply-ufunc.html) to make `dropna` to work with cupy arrays. I couldn't get it to work right away but putting a note here in case we want to revisit later.